src/main/java/de/tum/in/www1/artemis/service/user/UserService.java

```diff
@@ -474,6 +493,25 @@ protected void deleteUser(User user) {
         learningGoalProgressRepository.deleteAllByUserId(user.getId());
         exerciseHintActivationRepository.deleteAllByUser(user);
 
+        // deleting Posts belonging to the user and belonging to the conversations created by the user
+        var conversationsByUser = conversationRepository.findAllByCreator(user);
+        conversationsByUser.forEach(conversation -> postRepository.deleteAllByConversationId(conversation.getId()));
+        postRepository.deleteAllByAuthor(user);
+
+        // deleting Posts belonging to all OneToOneChats where the user is a participant
+        var conversationParticipantEntities = conversationParticipantRepository.findConversationParticipantsByUser(user);
+        var oneToOneChatsToDelete = new HashSet<OneToOneChat>();
+        conversationParticipantEntities.forEach(participant -> oneToOneChatsToDelete.addAll(oneToOneChatRepository.findAllByConversationParticipantsContaining(participant)));
+        oneToOneChatsToDelete.forEach(oneToOneChat -> postRepository.deleteAllByConversationId(oneToOneChat.getId()));
```

Comment:
```suggestion
        final Set<OneToOneChat> oneToOneChats = oneToOneChatRepository.findAllByParticipatingUser(user);
        postRepository.deleteAllByConversationIn(new HashSet<>(oneToOneChats));
```
with
```java
    @Query("""
            SELECT chat
            FROM OneToOneChat chat
                LEFT JOIN chat.conversationParticipants participants
                LEFT JOIN participants.user user
            WHERE user = :user
            """)
    Set<OneToOneChat> findAllByParticipatingUser(@Param("user") User user);
```
should reduce database calls here, too.